jcg_testadapter_commons/src/main/scala/AdapterOptions.scala

```diff
@@ -0,0 +1,10 @@
+case class AdapterOptions(
```

Comment:
This is better, but still completely Java specific, so it doesn't make much sense for other languages. Maybe we should just have this as a Map so we could supply different keys for different languages? If we want it to be safe, the map could be wrapped and checks implemented so only valid keys for the language are added, defaults are added where necessary, etc.

Reply:
So we wrap a Map of String -> Any which contains the options and validates if given keys match the given language?

Reply:
Something like that, yes. Could either validate after creation or enforce that only valid keys are added in the first place, e.g., by only constructing via language-specific factory methods or similar.